model/job-management/src/main/proto/beam_artifact_api.proto

```diff
@@ -67,6 +69,10 @@ message ArtifactMetadata {
 
 // A collection of artifacts.
 message Manifest {
+  enum Constants {
```

Comment:
I don't think `Manifest` is the right place for this.

The protobuf spec seems to support free-standing enums, but it looks like the constant enums elsewhere in Beam are wrapped in messages, causing redundant naming: `Api.BeamConstants.Constants.CONSTANT`. However, I'm not sure if that was deliberate, and if so what the reason might be.

Reply:
The constant is wrapped in the message (`Manifest`) and the reference is `ArtifactApi.Manifest.Constants.NO_ARTIFACTS_STAGED_TOKEN` https://github.com/apache/beam/pull/10285/files#diff-aed248327fcc4e847968c709c2fa5a1aR56

Reply:
I don't like that while `Constants` is scoped to `Manifest`, `Manifest` doesn't use `Constants`. Though admittedly this is a bit nit-picky.

Reply:
The proto code generator will place enums into the root namespace and not use the name of the enum as a namespace for some languages while in other languages like Java the enum's name is part of the namespace which is why they are wrapped inside a message.

We ran into this issue with the enums which used the same constant `UNSPECIFIED` where the code generation had issues.
See